controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -277,6 +272,10 @@ func (r *DataScienceClusterReconciler) ReconcileComponent(
 
 	r.Log.Info("Starting reconciliation of component: " + componentName)
 
+	enabled := component.GetManagementState(instance) == operatorv1.Managed
+	if componentName == componentsv1.ModelControllerComponentName {
+		enabled = instance.Spec.Components.ModelMeshServing.ManagementState == operatorv1.Managed || instance.Spec.Components.Kserve.ManagementState == operatorv1.Managed
+	}
```

Comment:
If I got this right, the model controller component would be installed in any case, regardless of the fact that either kserve or modelmesh are managed, but I don't think that should be what we aims to.

Reply:
yes, logic is done in GetManagedmentState() so this was some old code before GetManagedmentState was introduced.
` .status.conditions.type: ` should have `odh-model-controllerReady: true /false`  but `.status.installedCompolments` should not include it. WDYT?